src/OSPSuite.SimModelNative/src/FormulaChange.cpp

```diff
@@ -100,9 +103,17 @@ bool FormulaChange::PerformSwitchUpdate(double * y, double time)
 	if (_quantity->IsFormulaEqualTo(_newFormula))
 		return false; //same formula already set - no change
 
-	//really new formula - set new FORMULA
+	//---- really new formula - set new FORMULA
+
+	//first, add state variables used by the old formula to the set of changed RHS variables
+	_quantity->AppendUsedVariables(changedVariablesIndicesUsedInParameters);
+
+	//update formula
 	_quantity->SetFormula(_newFormula);
 
+	//now add state variables used by the NEW formula to the set of changed RHS variables
+	_quantity->AppendUsedVariables(changedVariablesIndicesUsedInParameters);
+
```

Comment:
2nd use case: old formula is overwritten by the new one.
Then state variables used both in the old and in the new formula are added to the list.